modules/core/arrow-core-data/src/main/kotlin/arrow/typeclasses/Traverse.kt

```diff
@@ -5,11 +5,519 @@ import arrow.core.Id
+ *
+ * ```kotlin
+ * import arrow.Kind
+ * import arrow.typeclasses.Applicative
+ * import arrow.typeclasses.Foldable
+ * import arrow.typeclasses.Functor
+ *
+ * interface Traverse<F> : Functor<F>, Foldable<F> {
+ *   fun <G, A, B> Kind<F, A>.traverse(AP: Applicative<G>, f: (A) -> Kind<G, B>): Kind<G, Kind<F, B>>
+ * }
+ * ```
+ *
+ * In our above example, `F` is `List`, and `G` is `Option`, `Either`, or `IO`. For the profile example, traverse says given a `List<User>` and a function `(User) -> IO<Profile>`, it can give you a `IO<List<Profile>>`.
+ *
+ * Abstracting away the `G` (still imagining `F` to be `List`), [traverse] says given a collection of data, and a function that takes a piece of data and returns an value `B` wrapped in a container `G`, it will traverse the collection, applying the function and aggregating the values (in a `List`) as it goes.
+ *
+ * In the most general form, `Kind<F, A>` is some sort of context `F` which may contain a value `A` (or several). While `List` tends to be among the most general cases, there also exist [Traverse] instances for `Option`, `Either`, and `Validated` (among others).
+ *
+ * ### When to use Traverse over Foldable
+ *
+ * Even though, [Foldable] and [Traverse] are related, because both 'reduce their values to something', it is not obvious why to consider [Traverse] over [Foldable].
+ *
+ * Here is one example:
+ *
+ * ```kotlin:ank:playground
+ * import arrow.core.MapK
+ * import arrow.core.Option
+ * import arrow.core.Some
+ * import arrow.core.extensions.option.applicative.applicative
+ * import arrow.core.fix
+ * import arrow.core.k
+ *
+ * fun main(){
+ *   //sampleStart
+ *   val map: MapK<String, Int> = mapOf("one" to 1, "two" to 2).k()
+ *
+ *   val optionMapK: Option<MapK<String, String>> = map.traverse(Option.applicative()) { Some("$it") }.fix()
+ *
+ *   // val optionMapKBoilered = map.foldLeft(Some(emptyMap())) { acc: Option<MapK<String, String>>, i: Int ->
+ *   //   acc.fold({ emptyMap() }, { /*Some logic to retrieve the key of a value, transform it and add it to the accumulated Map*/ })
+ *   // }
+ *   //sampleEnd
+ *   println(optionMapK)
+ * }
+ * ```
+ *
+ * Both methodologies try to attain the same thing, but what [Foldable] lacks is that it solely drills down to it's `A` here `Int` and does not preserve it's shape `F` - `MapK`. Resulting in a tiresome implementation, where we need to come up with an algorithm to resolve a key of a given value in the `Map` - let alone that this algorithm is not universal over any given `Map`.
+ *
+ * This is where [Traverse] shines, whenever you care about the Output `B` from `(A) -> B` and the existing shape of `F` you may use [traverse].
+ *
+ * Additionally, you're able to wrap your context `F` within a `G`. That is, may among others, the reason why [Traverse] is strictly more powerful than [Foldable].
+ *
+ * However, for certain computations [traverse] may also be simplified to a `map`:
```

Comment:
I don't think anyone would do this in the wild, probably better to go the other way around and argue that `traverse` over the identity functor also defines map. Not sure tho

Reply:
I'll delete that section. Down below I already explain how one can define map with traverse using `Id`

Reply:
In fact I will add it down below to that example as you said :) Thanks so much @1Jajen1